### PR TITLE
feat: localize pin handling to keychain

### PR DESCRIPTION
**Pull request by estrattonbailey**
Fixes APP-####

## What changed (plus any additional context for devs)
This PR moves all Android pin handling into the keychain module itself. So anytime we read/write private data, we prompt for pin code.

We had _a lot_ of manual loading state handling that had to be hidden for Android so that the pin code sheet could show on top. Instead of adding even more of this bloat, I removed the loading state entirely 🤷  Android is much faster than it was a while back, I don't think it's a massive issue. And if it is, we can solve it in other ways.

### CODE REVIEW

File: src/keychain/index.ts [hunk 1]
```diff
@@ -71,10 +86,62 @@ export async function get(
...
       } catch (e: any) {
         switch (e.toString()) {
+          /*
+           * Can happen if the user initially had biometrics enabled, installed
+           * the app, created a wallet, etc, and THEN disabled biometrics.
+           *
+           * In this case, values previously saved privately (like seephrase)
+           * will fail because the library can't authenticate.
+           */
+          case 'Error: code: 11, msg: No fingerprints enrolled.': {
```
Comment (skylarbarrera) [inline src/keychain/index.ts:134]: wonder if we may want to log this in sentry to track how often its happening

File: src/redux/wallets.ts [hunk 2]
```diff
@@ -56,7 +55,7 @@ interface WalletsState {
   /**
    * The current loading state of the wallet.
    */
-  isWalletLoading: WalletLoadingState | null;
+  isWalletLoading: any;
```
Comment (skylarbarrera) [inline src/redux/wallets.ts:58]: curious what this is being used for still

**Review by skylarbarrera — APPROVED** [review]
🐐

File: src/redux/wallets.ts (hunk 2 above)
Comment (estrattonbailey) [reply src/redux/wallets.ts:58]: Just wanted to keep file changes down, but tbh might only be 1-3 more files.

Comment (estrattonbailey) [reply src/redux/wallets.ts:58]: Eh I don't wanna axe all that old portal stuff, I don't fully grok it yet

File: src/components/backup/BackupConfirmPasswordStep.js [hunk 3]
```diff
@@ -133,14 +133,10 @@ export default function BackupConfirmPasswordStep() {
...
-      passwordRef.current?.focus();
-      setIsWalletLoading(null);
-      DelayedAlert({ title: msg }, 500);
-    },
-    [setIsWalletLoading]
-  );
+  const onError = useCallback(msg => {
+    passwordRef.current?.focus();
+    DelayedAlert({ title: msg }, 500);
+  }, []);
```
Comment (jkadamczyk) [inline src/components/backup/BackupConfirmPasswordStep.js:139]: Can we add explicit type or explicit any here? React 18 doesn't like that and the bump PR is just around the corner

**Review by jkadamczyk — CHANGES_REQUESTED** [review]
broke backups for me

**Review by jkadamczyk — APPROVED** [review]
LGTM! Thanks for fixing it up!

**Pull request merged**